api/gateway/gateway.go

```diff
@@ -48,6 +53,7 @@ type Gateway struct {
 	remoteCert                   string
 	gatewayAddr                  string
 	apiMiddlewareEndpointFactory apimiddleware.EndpointFactory
+	proxy                        *apimiddleware.ApiProxyMiddleware
```

Comment:
I doesn't make sense to include both fields. `proxy` already contains `apiMiddlewareEndpointFactory`.

Reply:
This is for the configuration option to store it in the struct so we can then create the proxy, so it's hard to refactor. Instead, I can create an embedded config struct